tools/mapping/gen_mapping_files/gen_ESMF_mapping_file/create_ESMF_map.sh

```diff
@@ -315,17 +307,15 @@ fi
 #-------------------------------------------------------------------------------
 
 case $MACH in
-  ## yellowstone, geyser, caldera, or pronghorn
-  "cheyenne" | "yellowstone" | "geyser" | "caldera" | "pronghorn" )
-    # From tcsh, script will not find module command
-    # So check to see if module works, otherwise source an init file
-    module list > /dev/null 2>&1 || source /etc/profile.d/modules.sh
+  ## cheyenne, geyser, caldera, or pronghorn
+  "cheyenne" |  "geyser" | "caldera" | "pronghorn" )
     module purge
-    module load intel
     module load nco
     if [  $MACH == "cheyenne" ]; then
+      module load intel
```

Comment:
If this tool were to invoke the cime/tools/configure script maybe it wouldn't be restricted to only run on this limited set of machines?   The configure script would provide a env_machine_specific file which should load a proper esmf environment if one is defined for that system.

Reply:
I like this idea, but ran into two issues with it:

1. on cheyenne ` ../../../configure --mpilib mpi-serial` worked fine but specifying `mpt` did not:

```
$ ../../../configure --mpilib mpt
ERROR: Multiple matches
```

I get the same error with `--mpilib openmpi` and `--mpilib mpich2`. Is there another mandatory argument? Adding ``--macros-format`` and ``--compiler`` doesn't change the error message. (Just a note: ``--clean`` works, so this branch has your fix for that issue)

2. on pronghorn, I needed to remove trilinos to avoid this error:

```
$ ../../../configure --mpilib mpi-serial
ERROR: module command /glade/apps/opt/lmod/lmod/libexec/lmod python load ncarenv/1.0 ncarbinlibs/1.1 perlmods gmake/4.1 python all-python-libs git intel/15.0.3 mkl/11.1.2 trilinos/11.10.2 esmf esmf-7.0.0-ncdfio-uni-O netcdf/4.3.3.1 ncarcompilers/1.0 cmake/3.0.2 all-python-libs failed with message:
Lmod Error: Cannot find Trilinos package built for intel version: 15.0.3
```

But then it looks like changing `--mpilib` causes different problems:

```
$ ../../../configure --mpilib mpich2
ERROR: module command /glade/apps/opt/lmod/lmod/libexec/lmod python load ncarenv/1.0 ncarbinlibs/1.1 perlmods gmake/4.1 python all-python-libs git intel/15.0.3 mkl/11.1.2 esmf esmf-7.0.0-defio-mpi-O netcdf-mpi/4.3.3.1 pnetcdf/1.6.1 ncarcompilers/1.0 cmake/3.0.2 all-python-libs failed with message:
Lmod Warning: Did not find: netcdf-mpi/4.3.3.1 pnetcdf/1.6.1

Try: "module spider netcdf-mpi/4.3.3.1 pnetcdf/1.6.1"
```

`netcdf-mpi/4.5.0` looks to be the only module built with `intel`, except it doesn't actually work:

```
$ module purge
$ module load intel/15.0.3
$ module avail netcdf-mpi

------------------------------------------------------- /glade/apps/opt/modulefiles/ca/cdep/intel -------------------------------------------------------
   netcdf-mpi/4.5.0

------------------------------------------------------- /glade/apps/opt/modulefiles/pr/cdep/intel -------------------------------------------------------
   netcdf-mpi/4.5.0

------------------------------------------------------- /glade/apps/opt/modulefiles/ys/cdep/intel -------------------------------------------------------
   netcdf-mpi/4.5.0
$ ../../../configure --mpilib mpich2
ERROR: module command /glade/apps/opt/lmod/lmod/libexec/lmod python load ncarenv/1.0 ncarbinlibs/1.1 perlmods gmake/4.1 python all-python-libs git intel/15.0.3 mkl/11.1.2 esmf esmf-7.0.0-defio-mpi-O netcdf-mpi/4.5.0 ncarcompilers/1.0 cmake/3.0.2 all-python-libs failed with message:
Lmod Error: Cannot find netcdf-mpi-4.5.0 built for intel version: 15.0.3
```

So if I can remove trilinos from the yellowstone entry in `config_machines.xml` then I think I'm good to go with using `--mpilib mpi-serial`... but there are more issues to work out before CIME's configure lets us use parallel.